lib/iex/lib/iex/autocomplete.ex

```diff
@@ -265,24 +265,37 @@ defmodule IEx.Autocomplete do
   end
 
   defp get_module_funs(mod) do
-    if function_exported?(mod, :__info__, 1) do
-      if docs = Code.get_docs(mod, :docs) do
-        Enum.filter_map(docs, &has_content?/1, &elem(&1, 0))
-      else
-        mod.__info__(:macros) ++ (mod.__info__(:functions) -- [__info__: 1])
-      end
+    docs = Code.get_docs(mod, :docs) || []
+    Enum.map(docs, &elem(&1, 0)) ++ module_info_funs(mod)
```

Comment:
`module_info_funs` will not return macros defined in Erlang.
At least it does not seem to return `Kernel.SpecialForms` macros.

```
iex(4)> Kernel.SpecialForms.__info__(:functions)
[]
iex(5)> Kernel.SpecialForms.__info__(:macros)   
[]
iex(6)> Kernel.SpecialForms.__info__(:exports)
[__info__: 1, module_info: 0, module_info: 1]
iex(8)> Code.get_docs(Kernel.SpecialForms, :docs) |> Enum.map(&elem(&1, 0))
[%: 2, %{}: 1, &: 1, .: 2, ::: 2, <<>>: 1, =: 2, ^: 1, __CALLER__: 0,
 __DIR__: 0, __ENV__: 0, __MODULE__: 0, __aliases__: 1, __block__: 1, alias: 2,
 case: 2, cond: 1, fn: 1, for: 1, import: 2, quote: 2, receive: 1, require: 2,
 super: 1, try: 1, unquote: 1, unquote_splicing: 1, with: 1, {}: 1]
```

AFAIK `module_info_funs` should be enough for all other cases though.
Do you think it would be better to handle special forms differently?